File: DesignPatternsExercises/java/TemplateMethodDesignPattern/Exercise/CarTravel.java
public class CarTravel extends Travel {
    @Override
    public void setDates() {
        System.out.println("Setting dates...");
        this.startDate = "01/01/2020";
        this.endDate = "01/10/2020";
    }

    @Override
    public void setDestination() {
        System.out.println("Setting destination...");
        this.destination = "Miami, FL, USA";
    }

    @Override
    void setMeal() {
    }

    @Override
    void usePrivateVehicle() {
        System.out.println("Selecting vehicle...");
        this.vehicle = "Honda Freed";
    }

    @Override
    void bookTicket() {
    }

    @Override
    void displayTravelDetails() {
        System.out.print("Car Travel:");
        __________(3)__________.displayTravelDetails();
        if (vehicle != null)
            System.out.println("    Vehicle selected: " + vehicle);
    }

    @Override
    boolean usePublicTransport() {
        return false;
    }
}

File: DesignPatternsExercises/ruby/template_method_design_pattern/solution/car_travel.rb
require_relative 'travel'

# class CarTravel < __________(5)__________
class CarTravel < Travel
  def dates
    puts 'Setting dates...'
    @start_date = '01/01/2020'
    @end_date = '01/10/2020'
  end

  def destination
    puts 'Setting destination...'
    @destination = 'Miami, FL, USA'
  end

  def meal
  end

  def use_private_vehicle
    puts 'Selecting vehicle...'
    @vehicle = 'Honda Freed'
  end

  def book_ticket
  end

  def display_travel_details
    print 'Car Travel:'
    # __________(6)__________
    super
    puts "    Vehicle selected: #{@vehicle}" unless @vehicle.nil?
    puts
  end

  def use_public_transport?
    false
  end
end
